resources/api/reservation.py

```diff
@@ -508,6 +509,66 @@ def has_object_permission(self, request, view, obj):
         return obj.can_modify(request.user)
 
 
+
+class ReservationAuthenticationLevelPermission(permissions.BasePermission):
+    """
+    This class matches the authentication level on resource with the current
+    authentication backend used to authenticate the user. User authenticated with
+    higher level authentication can reserve resource with lower level authentication,
+    plus the same level as user's authentication level.
+
+    User's auth level       Reserveable resource with auth level
+      Strong          ->       strong, mid, weak, none
+      Mid             ->       mid, weak, none
+      Weak            ->       weak, none
+    """
+    message = ''
+    # TODO: Needs discussing with client which login method belongs to which level.
+    STRONG_AUTHENTICATION = ('suomifi')
+    MID_AUTHENTICATION = ('axiell_aurora')
```

Comment:
These should probably be tuples?
```suggestion
    STRONG_AUTHENTICATION = ('suomifi',)
    MID_AUTHENTICATION = ('axiell_aurora',)
```